src/components/Popover/src/Popover.vue

```diff
@@ -0,0 +1,216 @@
+<template>
+	<div style="display: contents;">
+		<!-- @slot Element that the popover is tied or 'tethered' to. Must be a single node. -->
+		<slot
+			name="tether"
+			v-bind="actionAPI"
+		/>
+
+		<portal
+			:selector="popoverApi.targetSelector"
+		>
+			<div>
+				<m-transition-fade-in @after-leave="destroyPopper">
+					<popover-instance
+						v-if="isOpen"
+						:tether-el="tetherEl"
+						:popper-config="popperConfig"
+						@popover-instance:new-popper="setPopper"
+					>
+						<!-- @slot Content that will appear in the floating popover -->
+						<slot name="content" />
+					</popover-instance>
+				</m-transition-fade-in>
+			</div>
+		</portal>
+	</div>
+</template>
+
+<script>
+import { throwError } from '@square/maker/utils/debug';
+import { MTransitionFadeIn } from '@square/maker/components/TransitionFadeIn';
+import { Portal } from '@linusborg/vue-simple-portal';
+import { PopoverConfigKey, PopoverAPIKey } from './keys';
+import PopoverInstance from './PopoverInstance.vue';
+import { generateRandomId } from './utils';
+
+const MAX_TETHER_VNODE = 1;
+const SKIDDING_OFFSET = 0;
+const DISTANCE_OFFSET = 8;
+
+const DEFAULT_MODIFIERS = [
+	{
+		name: 'offset',
+		options: { offset: [SKIDDING_OFFSET, DISTANCE_OFFSET] },
+	},
+	{
+		name: 'preventOverflow',
+		options: {
+			padding: 0,
+			altBoundary: true,
+		},
+	},
+	{
+		name: 'minWidth',
+		enabled: true,
+		phase: 'beforeWrite',
+		requires: ['computeStyles'],
+		fn({ state: { styles, rects } }) {
+			styles.popper.minWidth = `${rects.reference.offsetWidth}px`;
+		},
+		effect({ state: { elements } }) {
+			elements.popper.style.minWidth = `${elements.reference.offsetWidth}px`;
+		},
+	},
+];
+
+export default {
+	name: 'Popover',
+
+	components: {
+		Portal,
+		PopoverInstance,
+		MTransitionFadeIn,
+	},
+
+	inject: {
+		popoverApi: {
+			from: PopoverAPIKey,
+			default: () => throwError('No popover layer detected', 'Popover'),
+		},
+
+		popoverConfig: {
+			from: PopoverConfigKey,
+			default: {
+				config: undefined,
+			},
+		},
+	},
+
+	props: {
+		/**
+		 * Starting position of the popover. Not guaranteed if too close to overflow.
+		 */
+		placement: {
+			type: String,
+			default: 'bottom-start',
+			validator: (placement) => [
+				'auto', 'auto-start', 'auto-end',
+				'top', 'top-start', 'top-end',
+				'right', 'right-start', 'right-end',
+				'bottom', 'bottom-start', 'bottom-end',
+				'left', 'left-start', 'left-end',
+			].includes(placement),
+		},
+
+		/**
+		 * Popper.js modifier list. Overrides default configuration if present.
+		 */
+		modifiers: {
+			type: Array,
+			default: () => DEFAULT_MODIFIERS,
+		},
+	},
+
+	data() {
+		const vm = this;
+
+		return {
+			/**
+			 * This is used to track which popover is open. When opening, this
+			 * will be pushed into the popoverApi state as the opened popover.
+			 */
+			id: generateRandomId(),
```

Comment:
Creating a `randomId` util for this is not worth it.

You'll get guaranteed uniqueness by closing over an `id` variable and just incrementing it each time.

Reply:
@landondurnan thoughts?

Reply:
I agree that we don't need to pull in another dependency to generate a unique ID.

Reply:
+1 for removing this

Reply:
Moved to an incremental ID